extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleDtpInternalMetricExtension.java

```diff
@@ -19,12 +19,18 @@
 import org.datatransferproject.api.launcher.ExtensionContext;
 import org.datatransferproject.spi.service.extension.ServiceExtension;
 
+import java.io.IOException;
+
 public class GoogleDtpInternalMetricExtension implements ServiceExtension {
 
   @Override
   public void initialize(ExtensionContext context) {
+    try {
     context.registerService(
         DtpInternalMetricRecorder.class,
-        new GoogleDtpInternalMetricRecorder(context.getMonitor()));
+        GoogleDtpInternalMetricRecorder.getInstance());
```

Comment:
Is the static initialization required for a particular scenario or can we obtain dependencies and instantiate via a public constructor?